contrib/enfuse_pro_2.lua

```diff
@@ -0,0 +1,1050 @@
+	text = pref_addtags,
+	placeholder = "Enter tags, seperated by commas",
+	editable = true
+}
+local widget = dt.new_widget("box") {
+    orientation = "vertical",
+    label_align_options,
+    chkbtn_radial_distortion,
+    chkbtn_optimize_field,
+    chkbtn_optimize_image_center,
+    chkbtn_auto_crop,
+    chkbtn_distortion,
+    cmbx_grid_size,
+    cmbx_control_points,
+    cmbx_control_points_remove,
+    cmbx_correlation,
+    separator1,
+    separator1b,
+    label_enfuse_options,
+    cmbx_fusion_type,
+    chkbtn_image_variations,
+    sldr_exposure_weight,
+    sldr_saturation_weight,
+    sldr_contrast_weight,
+    sldr_exposure_optimum_weight,
+    sldr_exposure_width_weight, 
+    chkbtn_hard_masks,
+    chkbtn_save_masks,
+    cmbx_contrast_edge_scale,
+    cmbx_contrast_min_curvature,
+    cmbx_contrast_window_size,
+    button_save_preset,
+    button_load_preset,
+    separator2,
+    separator2b,
+    label_output_format,
+    cmbx_output_format,
+    label_path,
+    file_chooser_button_path,
+	chkbtn_source_location,
+    cmbx_existing_file,
+	cmbx_style,
+	chkbtn_cpytags,
+	entry_tag,
+}
+
+-- FUNCTION -- 
+local function GetFileName(full_path)
+	--[[Parses a full path (path/filename_identifier.extension) into individual parts
+	Input: Folder1/Folder2/Folder3/Img_0001.CR2
+	
+	Returns:
+	path: Folder1/Folder2/Folder3/
+	filename: Img_0001
+	identifier: 0001
+	extension: .CR2
+	
+	EX:
+	path_1, file_1, id_1, ext_1 = GetFileName(full_path_1)
+	]]
+	local path = string.match(full_path, ".*[\\/]")
+	local filename = string.gsub(string.match(full_path, "[%w-_]*%.") , "%." , "" ) 
+	local identifier = string.match(filename, "%d*$")
+	local extension = string.match(full_path, "%.%w*")
+    return path, filename, identifier, extension
+end
+local function truncate(x)
+      return x<0 and math.ceil(x) or math.floor(x)
+end 
+local function replace_comma_to_dot(s)
+	return string.gsub(s, "%,", ".")
+end
+local function file_exists(name)
+   local f=io.open(""..name.."","r")
+   if f~=nil then 
+       io.close(f) 
+       return true 
+   else 
+       return false 
+   end
+end
+local function remove_temp_files()
+	if dt.configuration.running_os == "windows" then
+		dt.print_log("Deleting temp files...")
+		dt.control.execute("del "..images_to_align)
+		dt.control.execute("del "..images_to_blend)
+		dt.print_log("Done deleting")
+	else
+		dt.print_log("Deleting temp files...")
+		dt.control.execute("rm "..images_to_align)
+		dt.control.execute("rm "..images_to_blend)
+		dt.print_log("Done deleting")
+	end
+	
+	--[[
+	dt.print_error("rm "..images_to_align)
+	result_to_align_images=dt.control.execute("rm "..images_to_align)
+	dt.print_error("rm "..tmppath.."/aligned_*.tif")
+	result_aligned_images=dt.control.execute("rm "..tmppath.."/aligned_*.tif")
+	]]
+end
+local function build_execute_command(cmd, args, file_list)
+	local result = false
+
+	if dt.configuration.running_os == "macos" then
```

Comment:
I was kind of afraid something like this would happen when I changed check_if_bin_exists.  I should probably change check_if_bin_exists to not add the open -W -a on MacOS and let each individual script add it if it's necessary since it doesn't work in every instance.

Reply:
Yea I was surprised to find this actually. You were the one who originally clue me in to needing the open -W -a, and reading literiture online it seems like that would be the proper way to do it, but I get alignimagestack doesn't count as an actual application, but more an executable...?

Also, when removing that from the command, I was not able to succesfully do string.gsub("open -W -a","") and I don't know why. It wouldn't replace. That's why I had to break it up into three individual statements. Any idea why that might be?

Reply:
I originally did this because Mac users were trying to run GIMP and when GIMP opened, the script finished running instead of waiting for GIMP to finish.  The commands you're running don't have a GUI, but don't run with "open -W -a".  So it seems to me the best choice is for me to take it out of the library and each script writer needs to be aware that on Macs they need to do "open -W -a" if they want to run a GUI command.  I'll make that change and submit a PR  To get gsub to work you needed to escape the spaces and dashes, i.e. string.gsub(mac_command, "open% %-W% %-a ", "").

Reply:
I made the change and submitted a PR.  I'll leave it overnight for comment, then merge it in the morning.

Reply:
Great, thanks for the update and explanation about gsub. I should have realized that was my problem...

Reply:
I merged the change to check_if_bin_exists() so you can take out the lines getting rid of "open -W -a"